chingu-voyages/v50-tier2-team-14
# Feature/Create cart component

# Description
Created cart component and functionality and linked it to menu


## Main changes explained:
- Created redux store to store cart and menu states
- Created redux actions for cart and menu: fetchMenuDtaa, addToCart, removeFromCart, removeAllItemsFromCart)
- Created cart components (cart, cartItem, increaseDecreaseAmount)
- Connected cart and menu to Redux store and added necessary functionality
…

## How to test:
1. check into current branch
2. do `npm install` and `npm start` to run this PR locally
3. go to localhost:3000
4. you will see a menu and a cart icon in the navbar:

![Screenshot 2024-08-02 at 2 49 46 PM](https://github.com/user-attachments/assets/8f760520-71a4-4565-862c-1216cf73c517)

5. play around with adding items, increasing/decreasing item count, removing items from cart
6. verify that cart icon in the navbar has a pink oval showing total  items count and this count updates correctly
7. verify that cart opens and closes upon clickin on cart icon 
8. verify that total price is calculated correctly

![Screenshot 2024-08-02 at 2 50 11 PM](https://github.com/user-attachments/assets/3ba43c25-fc5d-448a-a96e-e289e048da82)

9. verify that you don't see 'loading...' when returning back to menu (by clicking on the cart icon)
10. comment on any bugs or suggestions 
11. NOTE: Proceed to checkout button does not yet work.

## Note:
I have implemented most of the cart's design but not the menu design, so don't judge that.
